west: Move cmake and build functionality to zephyr and add a boards command (#15704)
zephyrproject-rtos/zephyr: doc/guides/west/build-flash-debug.rst

Building, Flashing and Debugging
################################

West provides several commands for building, flashing, and interacting with

> Zephyr programs running on a board: ``boards``, ``build``, ``flash``, ``debug``,

Author: Done

   .. contents::
      :local:

.. _west-boards:

Listing boards: ``west boards``
*******************************

> .. tip:: Run ``west boards -h`` for additional help.

Reviewer: I'd delete this since you've already mention ``-h`` below (no need to mention it twice)

The ``boards`` command can be used to list the boards that are supported by
Zephyr without having to resort to additional sources of information.

It can be run by typing::

  west boards

> Which will list all supported boards in a default format. If you prefer to

Reviewer: ```suggestion
This command lists all supported boards in a default format. If you prefer to
```

Author: Fixed

  whether ``-b`` is required, just try leaving it out. West will print an
  error if the option is required and was not given.

.. tip::

>   You can use the :ref:```west boards`` <west-boards>` command to list all

Reviewer: Nice try.  That's not going to work. (Check the [doc build artifacts for this PR](https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/15704/docs/guides/west/build-flash-debug.html#building-west-build))
```suggestion
  You can use the :ref:`west boards <west-boards>` command to list all
```

Author: Fixed
